docs/event_listeners.md

```diff
@@ -0,0 +1,235 @@
+# Event Listeners
+
+When building a blueprint, your application may require to listen to events. Events can be of any type, and handling those events is entirely up to your discretion.
+
+## Creating Event Listeners
+
+To create an event listener, begin by defining a struct or enum which will listen to events:
+
+```rust
+use gadget_sdk::event_listener::EventListener;
+use async_trait::async_trait;
+
+/// Define a simple event listener that ticks every MS milliseconds.
+pub struct Ticker<const MS: usize> {
+    additional_delay: u64,
+}
+```
+
+Then, define a context for the event listener:
+
+```rust
+#[derive(Copy, Clone)]
+pub struct MyContext {
+    pub additional_delay: u64,
+}
+```
+
+Next, implement `EventListener` for `Ticker`:
+
+```rust
+/// Implement the [`EventListener`] trait for the Ticker struct. [`EventListener`] has two type parameters:
+/// 
+/// - T: is the type of the event that the listener listens for. This can be anything that is Send + Sync + 'static.
+/// In this case, we are using [`Instant`], which is a timestamp.
+///
+/// - Ctx: is the context type that the listener receives when constructed. This can be anything that is Send + Sync + 'static,
+/// with the special requirement that it is the *first* listed additional parameter in the [`job`] macro (i.e., a parameter not
+/// in params(...)). In this case, we are using [`MyContext`].
```

Comment:
That seems like an odd restriction? Why not mark it with `#[context]` or check for a parameter named `context`?

Reply:
I would also like to find an improvement over this. For now, this is just what was agreed upon by @shekohex and I.

Reply:
We could go with the more explicit path here with adding `#[context]` but we are aiming for less noise which would improve readability.

```rust
#[job(id = 1, params(x, y), result(_)]
fn my_job(context: Context, x: u64, y: u64) -> Result<u64, Inflatable> { ... } 
```
vs
```rust
#[job(id = 1, params(x, y), result(_)]
fn my_job(#[context] context: Context, x: u64, y: u64) -> Result<u64, Inflatable> { ... } 
```